### PR TITLE
testing/grpc-java: new aport

### PR DESCRIPTION
**Pull request by wenerme**

### CODE REVIEW

**Comment by jirutka** [comment]
⚠️ grpc must be very thoroughly reviewed if it does not bundle some libs or CA certificates as grpc-ruby do (see https://github.com/jirutka/user-aports/tree/v3.7/user/ruby-grpc). ⚠️

**Comment by wenerme** [comment]
This package has no ca, only contain the protoc-gen-grpc-java

File: testing/grpc-java/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+pkgrel=0
+pkgdesc="The Java gRPC implementation. HTTP/2 based RPC"
+url="https://github.com/grpc/grpc-java"
+arch="all"
+license="Apache2"
+depends="openjdk8-jre"
+makedepends="openjdk8 protobuf-dev"
+source="$pkgname-$pkgver.tar.gz::https://github.com/grpc/grpc-java/archive/v$pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$pkgver"
+options="!check"
```
Comment (andypost) [inline testing/grpc-java/APKBUILD]: Please add comment why tests are missing?

File: testing/grpc-java/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Contributor: wener <wenermail@gmail.com>
+# Maintainer: wener <wenermail@gmail.com>
+pkgname=grpc-java
+pkgver=1.10.0
+pkgrel=0
+pkgdesc="The Java gRPC implementation. HTTP/2 based RPC"
+url="https://github.com/grpc/grpc-java"
+arch="all"
```
Comment (andypost) [inline testing/grpc-java/APKBUILD]: Maybe java is `noarch`?

File: testing/grpc-java/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Contributor: wener <wenermail@gmail.com>
+# Maintainer: wener <wenermail@gmail.com>
+pkgname=grpc-java
+pkgver=1.10.0
```
Comment (andypost) [inline testing/grpc-java/APKBUILD]: 1.11.0 is out

File: testing/grpc-java/APKBUILD (hunk 2 above)
Comment (wenerme) [reply testing/grpc-java/APKBUILD]: No, compiler contain cpp
https://github.com/grpc/grpc-java/blob/master/compiler/src/java_plugin/cpp/java_plugin.cpp

**Comment by wenerme** [comment]
close this pr, I thought this package is not merged, create another pr to upgrade version

**Comment by wenerme** [comment]
Why no check

To run the test, gradle will download protc from maven, the downloaded protoc will not work on alpine, by add `-Pprotoc=/usr/bin/protoc` to gradle test, will use local protoc binary.

The test also contain javalite relate tests, will download protoc-gen-javalite, this package is located at [google/protobuf](https://github.com/google/protobuf/blob/master/java/lite.md), currently not available on alpine(I will build this later), so, check the build when this package has been built.

**Comment by jirutka** [comment]
Please do not close PRs, just update them.

**Comment by wenerme** [comment]
@jirutka should I redo the upgrade 1.11.0 here, or use this pr from next upgrade ?

**Comment by jirutka** [comment]
Once you’ve closed the PR, you cannot go back.

**Comment by wenerme** [comment]
I will careful later new aport, so , for upgrade, now should I create a pr for a version or a branch for all later upgrade ?

**Comment by jirutka** [comment]
I’m quite confused now. This PR has been actually merged (e2fb05b5c3f92ba4d53a3ab908ba53720824c16a).

Upgrades should be done in separate PRs, I thought that you wanted to update PR that is not merged it.

**Pull request closed without merging**